src/NITTA/LuaFrontend.hs

```diff
@@ -1,474 +1,391 @@
-processStatement fn (LocalAssign names (Just rexp)) =
-    processStatement fn $ Assign (map VarName names) rexp
+        addToBuffer name valueString serialNumber = do
+            algBuilder@AlgBuilder{algVars, algBuffer, algStartupArgs} <- get
+            let value = LuaValueVersion{luaValueVersionName = name, luaValueVersionAssignCount = 0, luaValueVersionIsConstant = False}
+            put algBuilder{algBuffer = HashMap.insert name value algBuffer, algVars = HashMap.insert value [] algVars, algStartupArgs = HashMap.insert serialNumber (name, valueString) algStartupArgs}
+            return value
+addStartupFuncArgs _ _ = undefined
+
+--Lua language Stat structure parsing
+--LocalAssign
+processStatement _ (LocalAssign _names Nothing) = do
+    return $ fromString ""
+processStatement fn (LocalAssign names (Just exps)) =
+    processStatement fn $ Assign (map VarName names) exps
+--Assign
 processStatement fn (Assign lexps@[_] [Unop Neg (Number ntype ntext)]) =
     processStatement fn (Assign lexps [Number ntype ("-" <> ntext)])
-processStatement _fn (Assign lexps@[_] [n@(Number _ textX)]) = do
-    let outs@[name] = parseLeftExp lexps
-    diff <- renameVarsIfNeeded outs
-    AlgBuilder{algItems} <- get
-    case findConstant textX algItems of
-        Just Constant{cVar} ->
-            -- TODO: sould work correctly, see test test_lua_two_name_for_same_constant
-            error $
-                concat
-                    [ "using several names for one constant value ("
-                    , show cVar
-                    , " and "
-                    , show name
-                    , "), please, use only one name (temporal restriction)"
-                    ]
-        Nothing -> do
-            -- `expConstant` return a wrong name, the correct name will be established
-            -- after all statement was processed
-            void $ expConstant name n
-            flushBuffer diff []
-        _ -> error "internal error"
-
--- e.g. @n, d = a / b@, or @n, d = f()@
-processStatement _fn (Assign lexps [rexp])
-    | length lexps > 1 = do
-        let outs = parseLeftExp lexps
-        diff <- renameVarsIfNeeded outs
-        rightExp diff outs rexp
-        flushBuffer diff outs
-
--- e.g. @a = 1@ or @a, b = 1, 2@
-processStatement _fn st@(Assign lexps rexps)
-    | length lexps == length rexps = do
-        let outs = parseLeftExp lexps
-        diff <- renameVarsIfNeeded outs
-        zipWithM_ (rightExp diff) (map (: []) outs) rexps
-        flushBuffer diff outs
-    | otherwise = error $ "assignment mismatch: " ++ show st
--- recursive call of main function
+processStatement _ (Assign lexp [rexp]) = do
+    _ <- parseRightExp (map parseLeftExp lexp) rexp
+    return $ fromString ""
+processStatement startupFunctionName (Assign vars exps) | length vars == length exps = do
+    mapM_ (\(VarName (Name name), expr) -> processStatement startupFunctionName (Assign [VarName (Name (getTempAlias name))] [expr])) $ zip vars exps
```

Comment:
Check lua standart on non-obliviour statements like `a, b = b + 2, a + 1`.

Reply:
works as expected. Test with this scenario added.